Kubernetes/HOWTO-on-prem.md

```diff
@@ -1,53 +1,72 @@
 
-### Install CNI ###
-We also need to install the basic CNI binaries so that networking works. Download them from [here](https://github.com/containernetworking/plugins/releases) and copy them to `/opt/cni/bin/`.
+ Copy these to `~/kube/bin` (note the extra `/bin` compared to the above path; we actual need to use these during master configuration, rather than just copying them to the Windows node). The v1.8 Windows binary `kubectl.exe` is included in the `windows/` directory of this repo, but you can also build it here (setting `KUBE_BUILD_PLATFORMS=windows/amd64` and `WHAT=cmd/kubectl` as above).
+
+### Install CNI Plugins ###
+We also need to install the basic CNI plugin binaries so that networking works. Download them from [here](https://github.com/containernetworking/plugins/releases) and copy them to `/opt/cni/bin/`.
 
 ## Prepare the Master ##
 Copy the entire directory from [this repository](https://github.com/Microsoft/SDN/tree/master/Kubernetes/linux), which contains a lot of the setup scripts and Kubernetes files that are essential to this process. Check them all out to `~/kube/` and make the scripts executable. This entire directory will be getting mounted for a lot of the docker containers in future steps, so keep its structure the same as outlined in the guide. 
 
 We'll be creating the service cluster on the virtual subnet 11.0/16. This isn't configurable, so if you want to tweak this, it'll require manual modification.
 
 ### Certificates ###
-First, prepare the certificates that will be used for nodes to communicate inthe cluster. Run `generate-certs.sh` (in the `certs/` directory), passing **your** external IP as its (only) parameter -- this is likely whatever the IP address of your `eth0` interface, or whatever:
+First, prepare the certificates that will be used for nodes to communicate in the cluster. Create a `certs/` directory, then move & run `generate-certs.sh` passing **your** external IP as its (only) parameter -- this is likely whatever the IP address of your `eth0` interface is. You can acquire this through `ifconfig` or through:
+
+    $ ip addr show dev eth0
+
+if you already know the interface name. Then:
 
+    ~/kube $ mkdir certs && cd certs
+    ~/kube/certs $ mv ../generate-certs.sh .
     ~/kube/certs $ ./generate-certs.sh 10.123.45.67   # example! replace
 
 ### Allow Routing ###
 _This step may be optional, depending on whether or not you've configured your intended cluster subnet to be routable already_.
 
-Windows nodes will each grab a /24 from the cluster, so if your cluster CIDR is 192.168.0.0/16, a Windows node will use 192.168.0.1/24. Just run:
+Windows nodes will each grab a /24 from the cluster CIDR, so it's 192.168.0.0/16, the first Windows node will use 192.168.0.1/24 for its pods. Pass the first two octets of your clister CIDR to generate the routes:
 
     ~/kube $ generate-routes.sh 192.168
 

```

Comment:
This assumes that we have a contiguous "/16" available which is usually not the case in an L3-routed network topology. Is there a way to shorten this POD CIDR? For instance, I only have three /26s which I need to give explicitly to my three nodes. They are contiguous but they don't start at 0 (e.g. 10.10.185.0/26, 10.10.185.64/26, 10.10.185.128/26)